### PR TITLE
Jmprieur/improve owin

### PR DESCRIPTION
**Pull request by jmprieur**

### CODE REVIEW

File: src/Microsoft.Identity.Web.OWIN/ApiControllerExtensions.cs [hunk 1]
```diff
@@ -17,31 +17,47 @@ public static class ApiControllerExtensions
...
-        /// <param name="apiController"></param>
         /// <returns></returns>
-        public static IDownstreamRestApi? GetDownstreamRestApi(this ApiController apiController)
+        public static IDownstreamRestApi GetDownstreamRestApi(this ApiController _)
         {
-            return TokenAcquirerFactory.GetDefaultInstance().ServiceProvider?.GetService(typeof(IDownstreamRestApi)) as IDownstreamRestApi;
+            IDownstreamRestApi? downstreamRestApi = TokenAcquirerFactory.GetDefaultInstance().ServiceProvider?.GetService(typeof(IDownstreamRestApi)) as IDownstreamRestApi;
+            if (downstreamRestApi == null)
+            {
+                throw new ConfigurationErrorsException("Cannot get IDownstreamRestApi. Did you add services.AddMicrosoftGraph() in Startup_Auth.cs? See https://aka.ms/ms-id-web/owin");
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web.OWIN/ApiControllerExtensions.cs:54]: >AddMicrosoftGraph [](http://example.com/codeflow?start=108&length=17)

You mean AddDownstreamRestAPI?

File: src/Microsoft.Identity.Web.OWIN/ControllerBaseExtensions.cs [hunk 2]
```diff
@@ -16,31 +17,47 @@ public static class ControllerBaseExtensions
...
-        /// <param name="controllerBase"></param>
         /// <returns></returns>
-        public static IDownstreamRestApi? GetDownstreamRestApi(this ControllerBase controllerBase)
+        public static IDownstreamRestApi GetDownstreamRestApi(this ControllerBase _)
         {
-            return TokenAcquirerFactory.GetDefaultInstance().ServiceProvider?.GetService(typeof(IDownstreamRestApi)) as IDownstreamRestApi;
+            IDownstreamRestApi? downstreamRestApi = TokenAcquirerFactory.GetDefaultInstance().ServiceProvider?.GetService(typeof(IDownstreamRestApi)) as IDownstreamRestApi;
+            if (downstreamRestApi == null)
+            {
+                throw new ConfigurationErrorsException("Cannot get IDownstreamRestApi. Did you add services.AddMicrosoftGraph() in Startup_Auth.cs? See https://aka.ms/ms-id-web/owin");
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web.OWIN/ControllerBaseExtensions.cs:54]: >AddMicrosoftGraph [](http://example.com/codeflow?start=108&length=17)

Get "DownstreamRestApi"

File: src/Microsoft.Identity.Web.OWIN/ControllerBaseExtensions.cs [hunk 3]
```diff
@@ -16,31 +17,47 @@ public static class ControllerBaseExtensions
...
-        /// <param name="controllerBase"></param>
         /// <returns></returns>
-        public static IAuthorizationHeaderProvider? GettAuthorizationHeaderProvider(this ControllerBase controllerBase)
+        public static IAuthorizationHeaderProvider GetAuthorizationHeaderProvider(this ControllerBase _)
         {
-            return TokenAcquirerFactory.GetDefaultInstance().ServiceProvider?.GetService(typeof(IAuthorizationHeaderProvider)) as IAuthorizationHeaderProvider;
+            IAuthorizationHeaderProvider? headerProvider = TokenAcquirerFactory.GetDefaultInstance().ServiceProvider?.GetService(typeof(IAuthorizationHeaderProvider)) as IAuthorizationHeaderProvider;
+            if (headerProvider == null)
+            {
+                throw new ConfigurationErrorsException("Cannot get GraphServiceClient. Did you create an OwinTokenAcquirerFactory in Startup_Auth.cs?. See https://aka.ms/ms-id-web/owin");
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web.OWIN/ControllerBaseExtensions.cs:40]: >GraphServiceClient [](http://example.com/codeflow?start=67&length=18)

AuthorizationHeaderProvider?

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: src/Microsoft.Identity.Web.OWIN/ApiControllerExtensions.cs (hunk 1 above)
Comment (jmprieur) [reply src/Microsoft.Identity.Web.OWIN/ApiControllerExtensions.cs]: Done. Thanks @jennyf19

File: src/Microsoft.Identity.Web.OWIN/ControllerBaseExtensions.cs (hunk 2 above)
Comment (jmprieur) [reply src/Microsoft.Identity.Web.OWIN/ControllerBaseExtensions.cs]: Done. Thanks @jennyf19

File: src/Microsoft.Identity.Web.OWIN/ControllerBaseExtensions.cs (hunk 3 above)
Comment (jmprieur) [reply src/Microsoft.Identity.Web.OWIN/ControllerBaseExtensions.cs]: Done. Thanks @jennyf19

**Pull request merged**